docs/developer/README.md

```diff
@@ -43,27 +43,96 @@ the `http://` version of an url, but Chrome will not let you:
+5. Under **Keycloak Realm > Clients** ([http://localhost:8080/admin/master/console/#/dummy-realm/clients](http://localhost:8080/admin/master/console/#/dummy-realm/clients)), import the client configuration using the [client config file](https://github.com/Aam-Digital/ndb-setup/blob/master/keycloak/client_config.json).
+6. In the new realm, create a user and assign relevant roles.
+
+### Step 3: Set Up CouchDB
+7. Access CouchDB at [http://localhost:5984/_utils/#database/app/_all_docs](http://localhost:5984/_utils/#database/app/_all_docs).
+8. Create a new database name as `app`.
+9. Add a document of type **Config:CONFIG_ENTITY** to the `app` database (e.g., from [dev.aam-digital.net CouchDB instance](https://dev.aam-digital.net/db/couchdb/_utils/#database/app/Config%3ACONFIG_ENTITY)).
+#### Note: If you get an error while adding a document (eg. document update conflict warning) remove the "_rev": "value".
+
+### Step 4: Start the Backend
+9. Retrieve the `public_key` for **dummy-realm** from [http://localhost:8080/realms/dummy-realm](http://localhost:8080/realms/dummy-realm) and add it to the `.env` file for the replication backend as `JWT_PUBLIC_KEY`.
+10. Start the replication backend:
+```shell
+    npm run start:dev
+```
+
+### Step 5: Start the Frontend
+11. Update `environment.ts` or `assets/config.json` with the following settings:
+```
+    "session_type": "synced",
+    "demo_mode": false
+```
+12. Start the frontend:
+```shell
+    npm start
+```
+
+### Accessing the Local Environment
+- Frontend App: [http://localhost:4200](http://localhost:4200)
+- Replication Backend: [http://localhost:3000](http://localhost:3000)
+- Keycloak: [http://localhost:8080](http://localhost:8080)
+- CouchDB: [http://localhost:5984](http://localhost:5984)
+
+---
+
+## Minimal Setup (Without Docker)
+
+For a basic setup without Docker, follow these steps.
+
+### Step 1: Set Up CouchDB
+1. Install CouchDB on Mac:
+```shell
+   brew install couchdb
+```
+2. Configure CouchDB:
+```shell
+   nano /opt/homebrew/etc/local.ini
+```
+3. Restart CouchDB:
+```shell
+   brew services restart couchdb
+```
+4. Access CouchDB at [http://localhost:5984/_utils/#](http://localhost:5984/_utils/#).
+
+### Step 2: Create Databases
+1. Create the following databases in CouchDB:
+- `app`
+- `app-attachments`
+- `notification-webhook`
+- `Report-calculation`
+
+3. In the `app` database, add these configuration documents:
+- **Config:CONFIG_ENTITY**
+- **Config:Permissions**
+
+**Note**: Please reach out to the tech team to get the specific details for these configuration documents.
+
+### Step 3: Set Up Replication Backend
+- Follow the project setup instructions provided in the [Backend Setup Guideline](https://github.com/Aam-Digital/replication-backend/blob/master/README.md).
+
+Once completed, you’ll have a minimal local environment without Docker.
```

Comment:
Does this one actually work, @Abhinegi2 ? Or does this run into the issue of data not syncing to the local CouchDB because authentication is missing?

Reply:
@sleidig  It works, but I think anyone working in the app needs to sync with the database at least once, so this might not be very helpful. Can I remove this? However, for just fetching the data—like we do in public forms—it can still retrieve the data successfully.

Reply:
if the normal create/update entities is not supported, then I think this option is more confusing than helpful. Let's remove it if that is the only case